core/src/main/java/org/apache/accumulo/core/client/admin/TableOperations.java

```diff
@@ -427,13 +427,8 @@ void compact(String tableName, Text start, Text end, List<IteratorSetting> itera
       throws AccumuloSecurityException, TableNotFoundException, AccumuloException;
 
   /**
-   * Starts a full major compaction of the tablets in the range (start, end]. If the config does not
-   * specify a compaction strategy, then all files in a tablet are compacted. The compaction is
-   * performed even for tablets that have only one file.
-   *
-   * <p>
-   * Only one compact call at a time can pass iterators and/or a compaction strategy. If two threads
-   * call compaction with iterators and/or a compaction strategy, then one will fail.
+   * Starts a full major compaction of the tablets in the range (start, end]. Files to be compacted
+   * will be selected by the compaction service configured on the table.
```

Comment:
There is a new plugin that a user can configure via compactionConfig.

https://github.com/apache/accumulo/blob/main/core/src/main/java/org/apache/accumulo/core/client/admin/compaction/CompactionSelector.java#L42

https://github.com/apache/accumulo/blob/1224e94e2651de3331aee542f59b4089c1b223b1/core/src/main/java/org/apache/accumulo/core/client/admin/CompactionConfig.java#L188

So could write something like the following

```
If the CompactionConfig object does not specify a CompactionSelector then all file in each tablet will be compacted.
```
Unless this behavior has changed, may be good to leave the following info and update it.

```
 If two threads call compaction with iterators and/or a compaction strategy, then one will fail.
```

Reply:
I can put together a suggested edit comment.